unittests/test_check_filters.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+
+import reframe.core.runtime as rt
+import reframe.frontend.check_filters as filters
+import unittests.fixtures as fixtures
+from reframe.core.systems import SystemPartition
+from reframe.core.pipeline import RegressionTest
+from reframe.utility.typecheck import Str
+
+class TestCheckFilters(unittest.TestCase):
+    def setUp(self):
+        #TODO: use correct initialization
+        #check 1
+        self.check1 = RegressionTest()
+        self.check1.name = 'check1'
+        self.check1.tags = {'a', 'b', 'c', 'd'}
+        self.check1.valid_prog_environs = ['env1', 'env2']
+        self.check1.valid_systems = ['testsys:gpu', 'testsys2:mc']
+        self.check1.num_gpus_per_node = 1
+        #check 2
+        self.check2 = RegressionTest()
+        self.check2.name = 'check2'
+        self.check2.tags = {'x', 'y', 'z'}
+        self.check2.valid_prog_environs = ['env3']
+        self.check2.valid_systems = ['testsys:mc', 'testsys2:mc']
+        self.check2.num_gpus_per_node = 0
```

Comment:
You should not individual tests here. You should have a list of tests instead:

```python
def create_check(attrs):
    ret = RegressionTest()
    for k, v in attrs.items():
        setattr(ret, k, v)

    return ret

self.checks = [create_check({'name': 'check1', ...}), create_check({'name': 'check2', ...})]
```

All subsequent unit tests should act on the list of checks `self.checks`. See example in my comments below.